pulp_rpm/app/models.py

```diff
@@ -1119,9 +1118,18 @@ class Image(Model):
     distribution_tree = models.ForeignKey(
         DistributionTree, on_delete=models.CASCADE, related_name='images'
     )
-    artifact = models.ForeignKey(
-        Artifact, on_delete=models.PROTECT, related_name='+'
-    )
+
+    @property
+    def artifact(self):
+        """
+        Returns artifact object.
+        """
+        content_artifact = ContentArtifact.objects.filter(
+            content=self.distribution_tree,
+            relative_path=self.path,
+        ).get()
```

Comment:
If this property is fetched while the DistributionTree was sync'd with `policy=on_demand` I think this will fail here.

What motivates having this here?

Reply:
I did it for linking `Image` and `Artifact`:
```
"images": [
                {
                    "name": "initrd",
                    "path": "images/pxeboot/initrd.img",
                    "platforms": "x86_64, xen",
                    "artifact": {
                        "_href": "/pulp/api/v3/artifacts/d6ddfc54-fe44-485d-9c3a-834a35aa8675/",
                        "_created": "2019-09-03T16:06:16.263824Z",
                        "file": "artifact/56/47f05ec18958947d32874eeb788fa396a05d0bab7c1b71f112ceb7e9b31eee",
                        "size": 2097152,
                        "md5": "b2d1236c286a3c0704224fe4105eca49",
                        "sha1": "7d76d48d64d7ac5411d714a4bb83f37e3e5b8df6",
                        "sha224": "7841241795c51351bb1b7dc7d41ea855eddf4143e8925c227490c390",
                        "sha256": "5647f05ec18958947d32874eeb788fa396a05d0bab7c1b71f112ceb7e9b31eee",
                        "sha384": "6f71dee19ba3fbdc5c15e857c98727eb91c318321c1c8d5a716a6b5d1b0404acb2a62fd975562545701013ec7f99329f",
                        "sha512": "731859029215873fdac1c9f2f8bd25a334abf0f3a9e1b057cf2cacc2826d86b0c26a3fa920a936421401c0471f38857cb53ba905489ea46b185209fdff65b3b6"
                    }
                },
```